File: backend/app/schemas/bazi.py
"""
八字相关的Pydantic模型
"""
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field
from uuid import UUID


class BaziCalculateRequest(BaseModel):
    """八字计算请求"""
    name: str = Field(..., max_length=50, description="姓名")
    gender: str = Field(..., description="性别（男/女）")
    calendar: str = Field(..., description="日历类型（公历/农历）")
    year: int = Field(..., ge=1900, le=2100, description="年")
    month: int = Field(..., ge=1, le=12, description="月")
    day: int = Field(..., ge=1, le=31, description="日")
    hour: int = Field(..., ge=0, le=23, description="时")
    minute: int = Field(..., ge=0, le=59, description="分")
    birth_city: str = Field(..., description="出生城市")
    current_city: Optional[str] = Field(None, description="现居城市")


class BaziProfileResponse(BaseModel):
    """八字档案响应"""
    id: UUID
    user_id: UUID
    name: str
    gender: str
    birth_info: dict
    bazi_result: dict
    created_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/api/bazi.ts
/**
 * 八字相关API
 */
import { get, post, del } from './request'
import type { BaziProfile } from '@/types'

/**
 * 计算八字
 */
export function calculateBazi(data: {
  name: string
  gender: string
  calendar: string
  year: number
  month: number
  day: number
  hour: number
  minute: number
  birth_city: string
  current_city?: string
}) {
  return post<BaziProfile>('/api/v1/bazi/calculate', data)
}

/**
 * 获取八字档案列表
 */
export function getBaziProfiles() {
  return get<BaziProfile[]>('/api/v1/bazi/profiles')
}

/**
 * 获取八字档案详情
 */
export function getBaziProfile(id: string) {
  return get<BaziProfile>(`/api/v1/bazi/profiles/${id}`)
}

/**
 * 删除八字档案
 */
export function deleteBaziProfile(id: string) {
  return del(`/api/v1/bazi/profiles/${id}`)
}

